tensorrec/tensorrec.py

```diff
@@ -294,16 +298,25 @@ def _build_tf_graph(self, n_user_features, n_item_features):
             'tf_prediction_serial': self.tf_prediction_serial,
             'tf_interactions_serial': tf_interactions_serial,
         }
-        if self.loss_graph_factory.is_dense:
+        if self.loss_graph.is_dense:
             loss_graph_kwargs.update({
                 'tf_prediction': self.tf_prediction,
                 'tf_interactions': tf_interactions,
                 'tf_rankings': self.tf_rankings,
                 'tf_alignment': self.tf_alignment,
             })
+        if self.loss_graph.is_sample_based:
+            tf_sample_predictions = gather_sampled_item_predictions(
+                tf_prediction=self.tf_prediction, tf_sampled_item_indices=self.tf_sampled_item_indices
+            )
+            tf_sample_alignments = gather_sampled_item_predictions(
+                tf_prediction=self.tf_alignment, tf_sampled_item_indices=self.tf_sampled_item_indices
+            )
+            loss_graph_kwargs.update({'tf_sample_predictions': tf_sample_predictions,
+                                      'tf_sample_alignments': tf_sample_alignments})
 
         # Build loss graph
-        self.tf_basic_loss = self.loss_graph_factory().loss_graph(**loss_graph_kwargs)
+        self.tf_basic_loss = self.loss_graph().loss_graph(**loss_graph_kwargs)
```

Comment:
I know it's due to the refactor and makes better sense that the reference to the class be called loss_graph instead of loss_graph_factory, but 
`self.loss_graph().loss_graph(**)` 
Just feels a bit wonky, but not worth changing.

Reply:
Maybe `loss_graph().connect_loss_graph()`?